src/Payum/Paypal/Rest/Tests/PaypalRestGatewayFactoryTest.php

```diff
@@ -72,6 +72,38 @@ public function shouldAllowCreateGateway()
         $this->assertAttributeNotEmpty('extensions', $extensions);
     }
 
+    /**
+     * @test
+     */
+    public function shouldAllowCreateGatewayWithCustomConfig()
+    {
+        $factory = new PaypalRestGatewayFactory();
+
+        $givenConfig = [
+            'log.LogLevel' => 'DEBUG',
+            'mode' => 'live',
+            'log.FileName' => '/foo/bar.log',
+            'http.ConnectionTimeOut' => '10',
+        ];
+
+        $gateway = $factory->create([
+            'client_id' => 'cId',
+            'client_secret' => 'cSecret',
+            'config' => $givenConfig,
+        ]);
+
+        $apis = $this->readAttribute($gateway, 'apis');
+        $apiContext = null;
+        foreach ($apis as $api) {
+            if ($api instanceof \PayPal\Rest\ApiContext) {
+                $apiContext = $api;
+            }
+        }
+
+        $this->assertNotNull($apiContext);
+        $this->assertArraySubset($givenConfig, $apiContext->getConfig());
```

Comment:
`assertArraySubset` is deprecated in PHPUnit. The two arrays should be the same at this point AFAICS

```suggestion
        $this->assertSame($givenConfig, $apiContext->getConfig());
```

Reply:
Oh didn't realize it became deprecated , but your're right sadly.
I've replaced it with 

```
        $apiContextConfig = $apiContext->getConfig();
        foreach ($givenConfig as $k => $v) {
            $this->assertArrayHasKey($k, $apiContextConfig);
            $this->assertSame($v, $apiContextConfig[$k]);
        }
```

since the `ApiContext`'s config does have some default values (which are not required to be set via the factory), `config` and `$apiContext->getConfig()` are not necessarily the same.